components/suggest/src/db.rs

```diff
@@ -322,101 +361,163 @@ impl<'a> SuggestDao<'a> {
         let keyword_lowercased = &query.keyword.to_lowercase();
         let (keyword_prefix, keyword_suffix) = split_keyword(keyword_lowercased);
         let suggestions_limit = &query.limit.unwrap_or(-1);
-        let suggestions = self.conn.query_rows_and_then_cached(
-            "SELECT s.id, MAX(k.rank) AS rank, s.title, s.url, s.provider, s.score, k.keyword_suffix
-            FROM suggestions s
-            JOIN prefix_keywords k ON k.suggestion_id = s.id
-            WHERE k.keyword_prefix = :keyword_prefix AND
-                k.keyword_suffix BETWEEN :keyword_suffix AND :keyword_suffix || x'FFFF' AND
-                s.provider = :provider
-            GROUP BY s.id
-            ORDER BY s.score DESC, rank DESC
-            LIMIT :suggestions_limit",
-             named_params! {
-                ":keyword_prefix": keyword_prefix,
-                ":keyword_suffix": keyword_suffix,
-                ":provider": SuggestionProvider::Amo,
-                ":suggestions_limit": suggestions_limit,
-            },
-            |row| -> Result<Option<Suggestion>>{
-                let suggestion_id: i64 = row.get("id")?;
-                let title = row.get("title")?;
-                let raw_url = row.get::<_, String>("url")?;
-                let score = row.get::<_, f64>("score")?;
+        let suggestions = self
+            .conn
+            .query_rows_and_then_cached(
+                r#"
+                SELECT
+                  s.id,
+                  MAX(k.rank) AS rank,
+                  s.title,
+                  s.url,
+                  s.provider,
+                  s.score,
+                  k.keyword_suffix
+                FROM
+                  suggestions s
+                JOIN
+                  prefix_keywords k
+                  ON k.suggestion_id = s.id
+                WHERE
+                  k.keyword_prefix = :keyword_prefix
+                  AND k.keyword_suffix
+                  BETWEEN :keyword_suffix
+                  AND :keyword_suffix || x'FFFF'
```

Comment:
```suggestion
                  AND (k.keyword_suffix BETWEEN :keyword_suffix AND :keyword_suffix || x'FFFF')
```

I think this is a bit more readable, and matches how you formatted weather!